ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -605,12 +604,37 @@ const DataSetPage = fp.flow(withUserProfile(), withCurrentWorkspace(), withUrlPa
       this.setState({selectedDomainValuePairs: valuesSelected, dataSetTouched: true});
     }
 
-    get selectAll() {
-      return fp.isEmpty(this.state.selectedDomainValuePairs);
+    // Append newValueSets (values from API) to state's valueSets and to selectedDomainValuePairs.
+    // Set valuesLoading to false once the state is updated
+    updateValueSets(newValueSets) {
+      const {selectedDomainValuePairs, valueSets} = this.state;
+
+      newValueSets.map(newValueSet => {
+        newValueSet.values.items.map(value => {
+          selectedDomainValuePairs.push({domain: newValueSet.domain, value: value.value});
+        });
+      });
+      this.setState({
+        valueSets: valueSets.concat(newValueSets),
+        selectedDomainValuePairs: selectedDomainValuePairs,
+        valuesLoading: false
+      });
+    }
+
+    // Returns true if selected values set is empty or is not equal to the total values displayed
+    get partialOrNoneValuesSelected() {
```

Comment:
Grammar nitpick: partialOrNoValuesSelected reads more correctly. Alternatively, partialOrNoneSelected reads well too – just the phrase "noneValuesSelected" sounds not quite right.